mlrun/runtimes/pod.py

```diff
@@ -220,11 +237,26 @@ def preemption_mode(self, mode):
         self._preemption_mode = mode or mlconf.function_defaults.preemption_mode
         self.enrich_function_preemption_spec()
 
+    @property
+    def security_context(self) -> k8s_client.V1SecurityContext:
+        return self._security_context
+
+    @security_context.setter
+    def security_context(self, security_context):
+        self._security_context = transform_attribute_to_k8s_class_instance(
+            "security_context", security_context
+        )
+
     def to_dict(self, fields=None, exclude=None):
-        struct = super().to_dict(fields, exclude=["affinity", "tolerations"])
+        exclude = exclude or []
+        exclude = list(set(exclude + ["affinity", "tolerations", "security_context"]))
+        struct = super().to_dict(fields, exclude=exclude)
         api = k8s_client.ApiClient()
         struct["affinity"] = api.sanitize_for_serialization(self.affinity)
         struct["tolerations"] = api.sanitize_for_serialization(self.tolerations)
+        struct["security_context"] = api.sanitize_for_serialization(
+            self.security_context
+        )
```

Comment:
Maybe let's try to make this code more generic so adding the next field would be just a matter of adding a string to a list ?